openquake/hazardlib/gsim/si_2020.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2014-2021 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`SiEtAl2020SInter`
+               :class:`SiEtAl2020SSlab`
+"""
+import numpy as np
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+def get_base_term(trt, C):
+    """
+    Returns the constant term of the GMM.
+    Depends on tectonic type
+    """
+    return C["e"] +\
+        (C["d0"] if trt == const.TRT.SUBDUCTION_INTERFACE else C["d1"])
+
+
+def get_magnitude_scaling_term(C, imt, mag):
+    """
+    Returns the magnitude scaling term (Equation 3.16)
+    """
+    if imt.string in ("PGA", "PGV"):
+        m_ref = 8.3
+    else:
+        if imt.period < 2.0:
+            m_ref = 8.3
+        else:
+            m_ref = 7.5
+    fmag = np.where(
+        mag < m_ref,
+        C["a1"] * mag,
+        C["a1"] * mag + (C["a2"] - C["a1"]) * (mag - m_ref)
+    )
+    return fmag
+
+
+def get_depth_scaling_term(C, hypo_depth):
+    """
+    Returns the depth scaling term (Eqation 3.16)
+    """
+    return C["h"] * hypo_depth
+
+
+def get_anelastic_attenuation_term(C, rrup):
+    """
+    Returns the anelastic attenuation term (Eq. 3.15)
+    The period dependent coefficients are calculated and added to the
+    coefficients table.
+    """
+    return C["c_attn"] * rrup
+
+
+def get_moho_depth(rup):
```

Comment:
Is this an attribute of the source, path or site? In the first and latter case It shouldn't be difficult to add it.

Reply:
The moho depth corresponds to the Moho depth at the location of the hypocentre (or at least in the model it is the hypocentral depth that is compared against the moho depth). From what I can discern from the PEER report, the Moho depth corresponds specifically to the J-SHIS Moho model (www.j-shis.bosai.go.jp/map/). In theory it could be a straightforward thing to do to simply have a corresponding Moho map in a binary format and just retrieve the Moho depth based on the hypocentre longitude and latitude. This might also be generalisable in the sense that the user
could supply their own Moho maps if they were to attempt to apply this GMPE in other regions.